data/remote/src/main/java/com/chocolate/remote/data_source/DirectMessageRemoteDataSourceImpl.kt

```diff
@@ -0,0 +1,151 @@
+const val SENTAT = "sentAt"
+
+class DirectMessageRemoteDataSourceImpl @Inject constructor(
+    private val firebase: FirebaseFirestore
+) : DirectMessageRemoteDataSource {
+    override suspend fun getChatsByUserId(
+        userid: String,
+        currentOrgName: String
+    ): Flow<List<Chat>> {
+        return callbackFlow {
+            val listener = firebase.collection(TEAMIX).document(currentOrgName).collection(CHATS)
+                .whereArrayContains("members", userid)
+                .orderBy(LASTMESSAGEDATE)
+                .addSnapshotListener { doc, error ->
+                    if (error != null)
+                        throw TeamixException(error.message)
+                    else {
+                        val chats = doc?.map {
+                            val members = it.data["members"] as List<String>? ?: emptyList()
+                            val secondMember = members.find { it != userid } ?: ""
+                            Chat(
+                                id = it.data["id"] as String? ?: "",
+                                secondMember = secondMember,
+                                lastMessage = it.data["lastMessage"] as String? ?: "",
+                                lastMessageDate = it.getTimestamp("lastMessageDate")?.toDate()
+                                    ?: Date()
+                            )
+                        } ?: emptyList()
+                        trySend(chats)
+                    }
+                }
+            awaitClose { listener.remove() }
+        }
+    }
+
+    override suspend fun createGroup(userids: List<String>, currentOrgName: String): String {
+        val groupCreatedBefore = checkIfGroupCreatedBefore(userids, currentOrgName)
+        return if (groupCreatedBefore.isNotEmpty()) groupCreatedBefore else
+            suspendCoroutine { cont ->
+                val newdoc = firebase.collection(TEAMIX).document(currentOrgName)
+                    .collection(CHATS)
+                    .document()
+                newdoc.set(
+                    NewChat(
+                        id = newdoc.id,
+                        members = userids
+                    )
+                )
+            }
+    }
+
+    private suspend fun checkIfGroupCreatedBefore(
+        userids: List<String>,
+        currentOrgName: String
+    ): String {
+        return suspendCoroutine { cont ->
+            firebase.collection(TEAMIX).document(currentOrgName).collection(CHATS)
+                .whereEqualTo("members", userids)
+                .get()
+                .addOnSuccessListener { doc ->
+                    val chat = doc.map {
+                        it.data["id"] as String? ?: ""
+                    }
+                    val groupId = if (chat.isEmpty()) "" else chat.first()
+                    cont.resume(groupId)
+                }.addOnFailureListener {
+                    cont.resumeWithException(it)
+                }
```

Comment:
use `await` method and wrap the whole method's body with `tryToExecuteSuspendCall`